engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/UniquelyConstructibleBuiltin.java

```diff
@@ -29,6 +30,6 @@ protected void postInitialize() {
   }
 
   public final Atom newInstance(Object... params) {
-    return uniqueConstructor.newInstance(params);
+    return AtomNewInstanceNode.getUncached().newInstance(uniqueConstructor, params);
```

Comment:
I decided to hide `AtomConstructor.newInstance` as _package private_ and expose `AtomNewInstanceNode.getUncached()` instead. It was not clear that `AtomConstructor.newInstance` was slow (and only allocating `BoxedAtom`!). Now, when we follow standard Truffle conventions, it is clear `getUncached()` is the slow path version of `AtomNewInstanceNode.create()`. Functionally they are identical as `AtomConstructorTest` demonstrates.